controllers/nstemplateset/cluster_resources.go

```diff
@@ -117,7 +117,7 @@ func (r *clusterResourcesManager) ensure(logger logr.Logger, nsTmplSet *toolchai
 
 		// get all objects of the resource kind from the template (if the template is specified)
 		if tierTemplate != nil {
-			newObjs, err = tierTemplate.process(r.Scheme, username, retainObjectsOfSameGVK(clusterResourceKind.gvk))
+			newObjs, err = tierTemplate.process(r.Scheme, map[string]string{"USERNAME": username}, retainObjectsOfSameGVK(clusterResourceKind.gvk))
```

Comment:
We could use a const string instead of hardcoding "USERNAME" many times across the package. To make it less error prone :)

Reply:
done in 161ef06